### PR TITLE
Skip Auto-installation in Audit SCA scan if requested by user (Yarn, NPM, Go)

### PR DESCRIPTION
**Pull request by eranturgeman**
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----

This PR introduces a new ability that allows skip auto installation in a project/module that is not installed and not specifically forced to be installed (using provision of an install command)
This ability is currently available for Yarn and NPM only (and for Golang since no installation required in it).

The ability is usable through Frogbot using JF_SKIP_AUTO_INSTALL env var (https://github.com/jfrog/frogbot/pull/758)
In addition, a new flag was added to enable using this ability through the CLI: **--skip-auto-install**

Linked PR: https://github.com/jfrog/build-info-go/pull/277

Failure in Nuget/.NET tests is known

### CODE REVIEW

File: commands/audit/sca/npm/npm_test.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package npm
 
 import (
 	"encoding/json"
+	utils2 "github.com/jfrog/build-info-go/utils"
```
Comment (attiasas) [inline commands/audit/sca/npm/npm_test.go:5]:
```suggestion
	buildInfoUtils "github.com/jfrog/build-info-go/utils"
```

File: commands/audit/sca/npm/npm_test.go [hunk 2]
```diff
@@ -122,3 +124,21 @@ func TestIgnoreScripts(t *testing.T) {
 	_, _, err := BuildDependencyTree(params)
 	assert.NoError(t, err)
 }
+
+// This test checks that the tree construction is skipped when the project is not installed and the user prohibited installation
+func TestSkipBuildDepTreeWhenInstallForbidden(t *testing.T) {
```
Comment (attiasas) [inline commands/audit/sca/npm/npm_test.go:129]: Can you add a test we requested to skip but the project is already installed? in this case we don't expect an error

File: commands/audit/sca/yarn/yarn.go [hunk 3]
```diff
@@ -136,19 +137,23 @@ func configureYarnResolutionServerAndRunInstall(params utils.AuditParams, curWd,
...
 		err = fmt.Errorf("failed to check the existence of '%s' file: %s", filepath.Join(currentDir, yarn.YarnLockFileName), err.Error())
 		return
 	}
 
-	// We verify the project's installation status by examining the presence of the yarn.lock file and the presence of an installation command provided by the user.
-	// Notice!: If alterations are made manually in the package.json file, it necessitates a manual update to the yarn.lock file as well.
-	if len(installCommandArgs) > 0 || !yarnLockExits {
-		installRequired = true
+	if len(installCommandArgs) > 0 {
+		return true, nil
```
Comment (attiasas) [inline commands/audit/sca/yarn/yarn.go:152]:
```suggestion
	if len(installCommandArgs) > 0 {
	        if skipAutoInstall {
	               log.Warn("Conflict: Requested to skip install but provided an install command")
	        }
		return true, nil
```

maybe skip should take priority always? even if cmd is provided? I think we should consult product

File: commands/audit/scarunner.go [hunk 4]
```diff
@@ -75,6 +76,11 @@ func buildDepTreeAndRunScaScan(auditParallelRunner *utils.SecurityParallelRunner
 		// Get the dependency tree for the technology in the working directory.
 		treeResult, bdtErr := buildDependencyTree(scan, auditParams)
 		if bdtErr != nil {
+			var installForbiddenErr *biutils.ErrInstallForbidden
```
Comment (attiasas) [inline commands/audit/scarunner.go:79]: `ErrInstallForbidden` is a bit confusing, I would change it to `ErrProjectNotInstalled`

**Review by attiasas — CHANGES_REQUESTED** [review]
I would also add it as a flag option to `audit` - `--skip-install`. we will need this in the future for the integrated tools

File: commands/audit/sca/npm/npm_test.go (hunk 2 above)
Comment (eranturgeman) [reply commands/audit/sca/npm/npm_test.go:129]: added for yarn and npm

File: commands/audit/sca/yarn/yarn.go (hunk 3 above)
Comment (eranturgeman) [reply commands/audit/sca/yarn/yarn.go:152]: this is what was approved by product

**Comment by eranturgeman** [comment]
@attiasas Af for the comment you left - I will open a Jira ticket to add the flag, as this is not part of the current feature

File: cli/docs/flags.go [hunk 5]
```diff
@@ -229,8 +230,9 @@ var flagsMap = map[string]components.Flag{
 		"Set to false if you wish to not use the gradle or maven wrapper.",
 		components.WithBoolDefaultValue(true),
 	),
-	WorkingDirs: components.NewStringFlag(WorkingDirs, "A comma-separated list of relative working directories, to determine audit targets locations."),
-	OutputDir:   components.NewStringFlag(OutputDir, "Target directory to save partial results to.", components.SetHiddenStrFlag()),
+	WorkingDirs:     components.NewStringFlag(WorkingDirs, "A comma-separated list of relative working directories, to determine audit targets locations."),
+	OutputDir:       components.NewStringFlag(OutputDir, "Target directory to save partial results to.", components.SetHiddenStrFlag()),
+	SkipAutoInstall: components.NewBoolFlag(SkipAutoInstall, "Set to true to skip auto-install of dependencies in un-built modules. Currently supported for Yarn and NPM only."),
```
Comment (eranturgeman) [inline cli/docs/flags.go:235]: set as hidden flag for now

**Review by attiasas — APPROVED** [review]

**Pull request merged**